scripts/cnode-helper-scripts/cncli.sh

```diff
@@ -797,10 +807,203 @@ cncliPTsendslots() {
+  sed 's/"//g' >> "$tmpcsv"
+}
+
+runNextEpoch() {
+  getKoiosData
+  getCurrNextEpoch
+  echo "Processing next epoch: ${next_epoch}"
+  stake_param_next="--active-stake ${active_stake_mark} --pool-stake ${pool_stake_mark}"
+
+  ${CNCLI} leaderlog ${cncliParams} --consensus "${consensus}" --epoch="${next_epoch}" --ledger-set next ${stake_param_next} |
+  jq -r '[.epoch, .epochNonce, .poolId, .sigma, .d, .epochSlotsIdeal, .maxPerformance, .activeStake, .totalActiveStake] | @csv' |
+  sed 's/"//g' >> "$tmpcsv"
+}
+
+runPreviousEpochs() {
+  [[ -z ${KOIOS_API} ]] && return 1
+
+  local one_or_many="${EPOCH}"
+  if [[ -n "$1" && "$1" != "all" ]]; then
+    one_or_many="$1"
+  elif [[ "${subarg}" != "all" && -n "${subarg}" ]]; then
+    one_or_many="${subarg}"
+  fi
+
+  if ! pool_hist=$(curl -sSL -f "${KOIOS_API}/pool_history?_pool_bech32=${POOL_ID_BECH32}&_epoch_no=${one_or_many}" 2>&1); then
+    echo "ERROR: Koios pool_stake_snapshot history query failed."
+    return 1
+  fi
+
+  if ! epoch_hist=$(curl -sSL -f "${KOIOS_API}/epoch_info?_epoch_no=${one_or_many}" 2>&1); then
+    echo "ERROR: Koios epoch_stake_snapshot history query failed."
+    return 1
+  fi
+
+  pool_stake_hist=$(jq -r '.[].active_stake' <<< "${pool_hist}")
+  active_stake_hist=$(jq -r '.[].active_stake' <<< "${epoch_hist}")
+
+  echo "Processing previous epoch: ${one_or_many}"
+  stake_param_prev="--active-stake ${active_stake_hist} --pool-stake ${pool_stake_hist}"
+
+  ${CNCLI} leaderlog ${cncliParams} --consensus "${consensus}" --epoch="${one_or_many}" --ledger-set prev ${stake_param_prev} |
+  jq -r '[.epoch, .epochNonce, .poolId, .sigma, .d, .epochSlotsIdeal, .maxPerformance, .activeStake, .totalActiveStake] | @csv' |
+  sed 's/"//g' >> "$tmpcsv"
+
+  return 0
+}
+
+processAllEpochs() {
+  getCurrNextEpoch
+  IFS=' ' read -r -a epochs_array <<< "$EPOCHS"
+
+  for EPOCH in "${epochs_array[@]}"; do
+    if ! getConsensus "${EPOCH}"; then echo "ERROR: Failed to fetch protocol parameters for epoch ${EPOCH}.; return 1"; fi
+     if [[ "$EPOCH" == "$curr_epoch" ]]; then
+       runCurrentEpoch
+     elif [[ "$EPOCH" == "$next_epoch" ]]; then
+       runNextEpoch
+     else
+       runPreviousEpochs
+     fi
+  done
+
+  id=1
+  while IFS= read -r row; do
+    echo "$id,$row" >> "$csvfile"
+    ((id++))
+  done < "$tmpcsv"
+
+  sqlite3 "$BLOCKLOG_DB" <<EOF
+DELETE FROM epochdata;
+VACUUM;
+.mode csv
+.import '$csvfile' epochdata
+REINDEX epochdata;
+EOF
+
+  row_count=$(sqlite3 "$BLOCKLOG_DB" "SELECT COUNT(*) FROM epochdata;")
+  echo "$row_count rows have been loaded into epochdata table in blocklog db"
+  echo "~ CNCLI epochdata table load completed ~"
+
+  rm $csvfile $tmpcsv
+}
+
+processSingleEpoch() {
+ if isNumber "$1"; then
+     getCurrNextEpoch
+     IFS=' ' read -r -a epochs_array <<< "$EPOCHS"
+
+   if [[ ! " ${epochs_array[@]} " =~ "$1" ]]; then
```

Comment:
This logic will/can cause issues. First of, when matching against array this will be read as a long string with all epochs separated with space. Matching this against a specific number can cause issues. Like 12 will match both 12 and 122. 

Safer to just loop array and break on match. something like this:
```
unset matched
for epoch in "${epochs_array[@]}"; do
  [[ ${epoch} = "$1" ]] && matched=true && break
done
if [[ -n ${matched} ]]; then
  echo -e "No slots found in blocklog table for epoch ${EPOCH}.\n"; return 1
fi
```

Reply:
applied and works as expected just changed -n for -z in 
if [[ -n ${matched} ]]; then...

Reply:
ahh yeah, -z is indeed the correct switch :)